tests/unit/clients/test_gce.py

```diff
@@ -120,5 +121,5 @@ async def test_list_instances_retrieves_multiple_pages(
         assert expected_results == results
         requests = list(m.requests.keys())
         assert 2 == len(requests)
-        assert ('get', filter_url) == requests[0]
-        assert ('get', url_with_token) == requests[1]
+        assert ('get', yarl.URL(filter_url)) == requests[0]
+        assert ('get', yarl.URL(url_with_token)) == requests[1]
```

Comment:
It feels like we're testing the wrong things here. We don't really care about the data type, only that the correct urls were called, which we already kind of know, since we only mocked the two endpoints and the results wouldn't match if the mock response data wasn't used.

I would be fine dropping these asserts altogether, since they don't seem to give anything valuable, and seem more to be testing the internals of aioresponses. If you absolutely want to keep them, I would change them to compare strings, because the yarl URL type is a side effect of the mocking library, not what we expect about the GCE client.

Reply:
I didn't look at what the tests are doing, just getting them to pass.  This isn't testing the yarl.URL()s, those tuples are actually the keys in the requests dict.  It's true, though, that testing these isn't super valid...

Reply:
What I mean by testing the `yarl.URL` type is that, because the syntax is `assert expected == actual`, we're saying we expect and are testing that this is object is of the type `yarl.URL`, as opposed to something like `assert filter_url == str(requests[0][1])` which allows our expected value to more closely match what we actually expect.

However I don't necessarily want to suggest that as my other point was that it seems like `requests` is actually an internal detail of `aioresponses` since it changed in a minor version, doesn't appear in the public docs, the data structure is awkward to use, and the way we're using it (casting a dictionary keyset to list to test order in < Python 3.7) is a bit strange as well.